actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -363,7 +363,8 @@ def setup(context, options, block)
         @options = options
         @block   = block
 
-        @locals  = options[:locals] ? options[:locals].symbolize_keys : {}
+        @locals  = options[:locals] || {}
+        @locals.symbolize_keys! if @locals.respond_to?(:symbolize_keys!)
```

Comment:
Does it make more sense to do `options[:locals].to_h.symbolize_keys` instead?

Reply:
I've considered that also, but that has other side-effects on the object that was passed in as well.

Consider this:
```ruby
render partial: "customer", locals: ActionController::Parameters.new(customer: { name: 'David' }).permit!
```
if we were to call `.to_h.symbolize_keys` on the value that is passed it, `to_h` will convert it to `{"customer"=>{"name"=>"David"}}` and `symbolize_keys` will then convert it to `{customer: {"name"=>"David"}}`.

However, if the application was expecting the name using `:name` it would break.

Passing the object untouched allows us to keep the old behaviour. In the case of `AC:Parameters`, since it allows indifferent access, `:name` works if nothing messes with it.